src/overlays/actors/ovl_Bg_Haka_Trap/z_bg_haka_trap.c

```diff
@@ -40,36 +41,490 @@ const ActorInit Bg_Haka_Trap_InitVars = {
+                thisx->params = TRAP_GUILLOTINE_SLOW;
+                this->unk_16A = 1;
+            }
+
+            this->actionFunc = &func_80880484;
+        } else {
+            DynaPolyInfo_SetActorMove(&this->dyna, DPM_PLAYER);
+            thisx->flags |= 0x10;
+
+            if (thisx->params == TRAP_SPIKED_BOX) {
+                DynaPolyInfo_Alloc(&D_06009CD0, &sp2C);
+                this->actionTimer = 30;
+
+                if ((*temp_ptr) != 0) {
+                    this->actionFunc = &func_808808F4;
+                    D_80881014 = 0;
+                } else {
+                    D_80881014 = 1;
+                    this->actionFunc = &func_808806BC;
+                    thisx->velocity.y = 0.5f;
+                }
+
+                thisx->unk_80 = thisx->initPosRot.pos.y - 225.0f;
+                this->unk_16A = (s16)((thisx->unk_80 + 50.0f) - 25.0f);
+
+                this->colliderCylinder.dim.radius = 10;
+                this->colliderCylinder.dim.height = 40;
+            } else {
+                if (thisx->params == TRAP_SPIKED_WALL) {
+                    DynaPolyInfo_Alloc(&D_060081D0, &sp2C);
+                    thisx->initPosRot.pos.x -= 200.0f;
+                } else {
+                    thisx->initPosRot.pos.x += 200.0f;
+                    DynaPolyInfo_Alloc(&D_06008D10, &sp2C);
+                }
+
+                Collider_InitTris(globalCtx, &this->colliderSpikes);
+                Collider_SetTris(globalCtx, &this->colliderSpikes, thisx, &sTrisInit, &this->colliderSpikesItem);
+
+                this->colliderCylinder.dim.radius = 18;
+                this->colliderCylinder.dim.height = 115;
+
+                this->colliderCylinder.body.toucherFlags = this->colliderCylinder.body.toucherFlags;
+                this->colliderCylinder.body.toucherFlags |= 0x10;
+
+                this->actionFunc = &func_808801B8;
+            }
+
+            this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, sp2C);
+        }
+    } else {
+        this->actionTimer = 40;
+        this->actionFunc = &func_808809B0;
+        thisx->unk_F8 = 500.0f;
+    }
+
+    func_80061ED4(&thisx->colChkInfo, 0, &sColChkInfoInit);
+}
+
+void BgHakaTrap_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTrap* this = THIS;
+
+    if (thisx->params != TRAP_PROPELLER) {
+        if (thisx->params != TRAP_GUILLOTINE_SLOW) {
+            DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+            if ((thisx->params == TRAP_SPIKED_WALL) || (thisx->params == TRAP_SPIKED_WALL_2)) {
+                Collider_DestroyTris(globalCtx, &this->colliderSpikes);
+            }
+        }
+
+        Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    }
+
+    func_800F89E8(&this->unk_16C);
+}
+
+void func_8087FFC0(BgHakaTrap* this, GlobalContext* globalCtx) {
+    f32 cosine;
+    Vec3f sp28;
+    f32 sine;
+    f32 z_non_negative;
```

Comment:
We use camelCase over snake_case.
```suggestion
    f32 zNonNegative;
```

Reply:
fixed, thanks!